src/process/message_processor.go

```diff
@@ -1,48 +1,101 @@
 package process
 
 import (
+	"BGPAlert/analyze"
 	"BGPAlert/common"
-	"errors"
+	"BGPAlert/config"
 	"fmt"
+	"strconv"
 	"time"
 )
 
-// Constantly reads channel of messages and stores them in Window objects to send through windowChannel for analysis
-func ProcessBGPMessages(msgChannel chan common.BGPMessage, windowChannel chan common.Window) error {
-	var bucketMap = make(map[time.Time][]common.BGPMessage)
+// Constantly read messages from channel, build up windows with frequency maps, calling analysis on windows when full
+func ProcessBGPMessages(msgChannel chan common.BGPMessage, config *config.Configuration) error {
+	// Parse windowSize from config
+	var maximumBuckets int
+	// If windowSize can't be parsed to an int -> just default to 30
+	if windowSize, err := strconv.ParseInt(config.WindowSize, 10, 64); err != nil {
+		fmt.Println("Inputted config size is not a number, defaulting to 30")
+		maximumBuckets = 30
+	} else {
+		maximumBuckets = int(windowSize)
+	}
+
+	// Each bucket is 60s
+	parseDuration := 60 * time.Second
+	maximumTimespan := time.Duration(maximumBuckets*60) * time.Second
+
+	window := common.Window{
+		Filter:    "none",
```

Comment:
Should the filter be pulled from the config? (I am not sure, as with my previous comments this is a question haha)

Reply:
This is just so we can handle filtering in the future